Improve system-requirements documentation (#21702)
opf/openproject: docs/installation-and-operations/system-requirements/README.md

This is for a single server running OpenProject for up to 200 total users. Depending on your number of concurrent users,  these requirements might vary drastically.

## Database

OpenProject officially supports [PostgreSQL version 16](https://www.postgresql.org/) or above since [OpenProject 16.0.0](https://www.openproject.org/docs/release-notes/16-0-0/).

> PostgreSQL versions 13 - 15 are not officially supported, but MAY continue to work, but may result in incompatibilities and degraded performance in the future. If you are using one of these versions currently, we have a [migration guide on how to upgrade to PostgreSQL 17](../misc/migration-to-postgresql17/) and strongly recommend you to upgrade your DBMS, as there are significant performance improvements.

Reviewer: ```suggestion
PostgreSQL versions 13 - 15 are not officially supported, but MAY continue to work, but could result in incompatibilities and degraded performance in the future. If you are using one of these versions currently, we have a [migration guide on how to upgrade to PostgreSQL 17](../misc/migration-to-postgresql17/) and strongly recommend you to upgrade your DBMS, as there are significant performance improvements.
```

> PostgreSQL versions 13 - 15 are not officially supported, but MAY continue to work, but could result in incompatibilities and degraded performance in the future. If you are using one of these versions currently, we have a [migration guide on how to upgrade to PostgreSQL 17](../misc/migration-to-postgresql17/) and strongly recommend you to upgrade your DBMS, as there are significant performance improvements.

Reviewer: ```suggestion
PostgreSQL versions 13 - 15 are not officially supported, but MAY continue to work and could result in incompatibilities and degraded performance in the future. If you are using one of these versions currently, we have a [migration guide on how to upgrade to PostgreSQL 17](../misc/migration-to-postgresql17/) and strongly recommend you to upgrade your DBMS, as there are significant performance improvements.
```

### Overview of dependencies

Both the package and docker based installations will install and setup the the [Ruby runtime](https://www.ruby-lang.org/en/)  as well as the [Puma application server](https://puma.io/) that are required by OpenProject to run.

> For the [packaged installation](../installation/packages) and the [all-in-one docker container](../installation/docker#all-in-one-container) container, an [Apache](https://httpd.apache.org/) web server is installed, and a [PostgreSQL 17](https://www.postgresql.org/)  database.

Reviewer: ```suggestion
For the [packaged installation](../installation/packaged/) and the [all-in-one docker container](../installation/docker#all-in-one-container) container, an [Apache](https://httpd.apache.org/) web server and a [PostgreSQL 17](https://www.postgresql.org/) database are installed.
```

> 

Reviewer 2: ```suggestion
The all-in-one container only will additionally install [hocuspocus](https://github.com/opf/op-blocknote-hocuspocus) which is required for the real-time collaboration feature in OpenProject.
```

### Overview of dependencies

> Both the package and docker based installations will install and setup the the [Ruby runtime](https://www.ruby-lang.org/en/)  as well as the [Puma application server](https://puma.io/) that are required by OpenProject to run.

Reviewer: ```suggestion
Both the package and docker based installations will install and setup the the [Ruby runtime](https://www.ruby-lang.org/en/), as well as the [Puma application server](https://puma.io/) that are required by OpenProject to run.
```

For the [packaged installation](../installation/packaged/) and the [all-in-one docker container](../installation/docker#all-in-one-container) container, an [Apache](https://httpd.apache.org/) web server and a [PostgreSQL 17](https://www.postgresql.org/) database are installed.

> The all-in-one container only will additionally install [hocuspocus](https://github.com/opf/op-blocknote-hocuspocus) which is required for the real-time collaboration feature in OpenProject.

Reviewer: ```suggestion
The all-in-one container will only additionally install [hocuspocus](https://github.com/opf/op-blocknote-hocuspocus), which is required for the [real-time collaboration](../../user-guide/documents/#collaborative-editing) feature in OpenProject.
```

Reviewer 2: I agree with the link. Thanks for adding that! Not with the placement of "only" though. The idea here was to emphasize that only the AIO container adds this additional dependency whilst the package does not.

You can get that meaning from the new order too but it's not very obvious in my opinion.
